catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -534,28 +546,111 @@ public void init() {
         APPLICATION_JSON,
         (req, res) -> {
           String id = req.params(":id");
+          Map<String, Object> body = mapper.parser().parseMap(util.safeGetBody(req));
 
-          Map<String, Object> workspace =
-              JsonFactory.create().parser().parseMap(util.safeGetBody(req));
-
-          Metacard metacard = transformer.transform(workspace);
-          metacard.setAttribute(new AttributeImpl(Metacard.ID, id));
+          Metacard updated = transformer.transform(body);
+          updated.setAttribute(new AttributeImpl(Core.ID, id));
 
-          Metacard updated = updateMetacard(id, metacard);
-          return util.getJson(transformer.transform(updated));
+          Metacard stored = updateMetacard(id, updated);
+          return util.getJson(transformer.transform(stored));
         });
 
     delete(
         "/workspaces/:id",
         APPLICATION_JSON,
         (req, res) -> {
           String id = req.params(":id");
+          WorkspaceMetacardImpl workspace = workspaceService.getWorkspaceMetacard(id);
+
+          String[] queryIds = workspace.getQueries().toArray(new String[0]);
+
+          if (queryIds.length > 0) {
+            catalogFramework.delete(new DeleteRequestImpl(queryIds));
+          }
+
           catalogFramework.delete(new DeleteRequestImpl(id));
+
           subscriptions.removeSubscriptions(id);
           return ImmutableMap.of("message", "Successfully deleted.");
         },
         util::getJson);
 
+    get(
+        "/workspace/:id/queries",
+        (req, res) -> {
+          String workspaceId = req.params(":id");
+          WorkspaceMetacardImpl workspace = workspaceService.getWorkspaceMetacard(workspaceId);
+
+          List<String> queryIds = workspace.getQueries();
+
+          return util.getMetacardsWithTagById(queryIds, QUERY_TAG)
+              .values()
+              .stream()
+              .map(Result::getMetacard)
+              .map(queryTransformer::transformMetacardIntoMap)
+              .collect(Collectors.toList());
+        },
+        util::getJson);
+
+    post(
+        "/query",
+        APPLICATION_JSON,
+        (req, res) -> {
+          Map<String, Object> body = mapper.parser().parseMap(util.safeGetBody(req));
+
+          Metacard metacard = queryTransformer.transformJsonIntoMetacard(body);
+          Metacard saved = saveMetacard(metacard);
```

Comment:
Does this endpoint need to associate the new query with a workspace?

Reply:
Not anymore. Associations are going to be updated with `PUT /workspaces/:id`